File: Tarea5_BubbleSort_19-09-25/LEM_BubbleSort.js
function bubbleSort(arr) {
    let n = arr.length;
    for (let i = 0; i < n; i++) {
        let swap = false;
        for (let j = 0; j < n - i - 1; j++) {
            if (arr[j] > arr[j + 1]) {
                let temp = arr[j];
                arr[j] = arr[j + 1];
                arr[j + 1] = temp;
                swap = true;
            }
        }
        if (!swap) {
            break;
        }
    }
}

let arr = Array.from({ length: 10 }, () => Math.floor(Math.random() * 100) + 1);

console.log("Arreglo original:", arr);

bubbleSort(arr);

console.log("Arreglo ordenado:", arr);

File: Tarea5_BubbleSort_19-09-25/LEM_BubbleSort.cs
using System;

class Program
{
    static void BubbleSort(int[] arr)
    {
        int n = arr.Length;
        for (int i = 0; i < n; i++)
        {
            bool swap = false;
            for (int j = 0; j < n - i - 1; j++)
            {
                if (arr[j] > arr[j + 1])
                {
                    int temp = arr[j];
                    arr[j] = arr[j + 1];
                    arr[j + 1] = temp;
                    swap = true;
                }
            }
            if (!swap)
            {
                break; 
            }
        }
    }

    static void Main()
    {
        const int SIZE = 10;
        int[] arr = new int[SIZE];
        Random rnd = new Random();

        for (int i = 0; i < SIZE; i++)
        {
            arr[i] = rnd.Next(1, 101);
        }

        Console.WriteLine("Arreglo original: " + string.Join(" ", arr));

        BubbleSort(arr);

        Console.WriteLine("Arreglo ordenado: " + string.Join(" ", arr));
    }
}
